main.cpp

```diff
@@ -19,15 +19,33 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter): _rewriter(rewriter) {};
 
-    void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+    void run(const MatchFinder::MatchResult &Result) override 
+    {
+        const auto *expression = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        
+        auto range_replace = CharSourceRange::getCharRange(expression->getLParenLoc(),
+                             expression->getSubExprAsWritten()->getBeginLoc());
+        auto &src_manage = *Result.SourceManager;
+
+        auto type_name = Lexer::getSourceText(CharSourceRange::getTokenRange(

```

Comment:
Please describe what the `getSourceText()` function does.

Reply:
getSourceText() returns a string for the source that the range encompasses.
For my code the range is: CharSourceRange::getTokenRange(
                                              expression->getLParenLoc().getLocWithOffset(1),
                                              expression->getRParenLoc().getLocWithOffset(-1))